### ISSUE TITLE
[Accessibility]: Commits page - Losing keyboard focus

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#34970]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34423

### Problem description

Two issues found:

**Commit message focus styles are cut off and not properly visible**

<img width="483" alt="image" src="https://user-images.githubusercontent.com/9516420/166906991-c90edf8a-003d-485b-a6aa-20c1e52bad1d.png">

**Show more button causes focus to be lost**
Clicking "Show more" means focus is lost on the page. Related to https://github.com/sourcegraph/sourcegraph/issues/34670

<img width="938" alt="image" src="https://user-images.githubusercontent.com/9516420/166907509-ea083c77-23a1-4104-9515-56b3e8df556f.png">

### Expected behavior

**Commit message focus styles are cut off and not properly visible**
Focus outline should fully surround the commit message text

**Show more button causes focus to be lost**
Focus should remain on the show more button. I believe this bug happens because the "Show more" button is removed once clicked. We should aim to keep it in the DOM, and disable it if there's no more results

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
